server/src/services/completion/onCompletion.ts

```diff
@@ -549,3 +559,71 @@ const getFullName = (node: Node | undefined): string | undefined => {
     return node.name;
   }
 };
+
+const getNatspecCompletion = (
+  documentAnalyzer: ISolFileEntry,
+  document: TextDocument,
+  position: VSCodePosition
+) => {
+  // Check that the current line has the natspec string
+  const searchString = "/** */";
+  const lineText = document.getText({
+    start: { line: position.line, character: 0 },
+    end: { line: position.line + 1, character: 0 },
+  });
+
+  if (!lineText.includes(searchString)) {
+    return null;
+  }
+
+  // Find the first function definition after current position
+  const currentOffset = document.offsetAt(position);
+  let nextFunction: FunctionDefinition | undefined;
+
+  parser.visit(documentAnalyzer.analyzerTree.tree.astNode, {
+    FunctionDefinition(node) {
+      if (!node.range || node.range[0] < currentOffset) {
+        return;
+      }
+      if (
+        nextFunction === undefined ||
+        node.range[0] < nextFunction.range![0]
+      ) {
+        nextFunction = node;
+      }
+    },
+  });
+
+  if (nextFunction === undefined) {
+    return null;
+  }
+
+  // Autocomplete with @dev, @param and @return tags
+  const range = {
+    start: position,
+    end: position,
+  };
+
+  let text = `\n * @dev Function description\n`;
+
+  for (const param of nextFunction.parameters) {
+    text += ` * @param ${param.name} parameter description\n`;
+  }
+
+  for (const _param of nextFunction.returnParameters ?? []) {
+    text += ` * @return return value description\n`;
```

Comment:
Should be `@return ${param.name} value description;`

Reply:
Right! forgot about named returns